app/api/chat/route.ts

```diff
@@ -36,7 +37,16 @@ export async function POST(req: Request) {
       previousMessages = await getLangchainMemories(room_id, 100);
     }
 
-    const currentMessage = new HumanMessage(question);
+    let messageText = question;
+    
+    if (room_id) {
+      const knowledgeContent = await getKnowledgeBaseContext(room_id);
+      if (knowledgeContent) {
+        messageText += `\n\n-----ARTIST KNOWLEDGE BASE-----\n${knowledgeContent}\n-----END KNOWLEDGE BASE-----\n\nPlease use the information from the knowledge base if relevant to answering my question.`;
```

Comment:
why are you appending this to the input text rather than to the system messages?